### PR TITLE
Switch macOS CI to use LLVM/clang

### LINKED ISSUE #293: Switch macOS CI compiler to Clang
I propose we switch the compiler of the macOS builds from `gcc` to `clang`. From my experience from CI, `clang` tends to be less painful to deal with. Plus, we prevent further scenarios where our code is only compiling in `gcc` by enforcing at all times it also works for `clang`.

In the case of #240, `gcc` doesn't seem to understand that some source code is actually Objective-C and thus fails to compile. Switching to `clang` fixed [the problem](https://github.com/rsubtil/cubos/actions/runs/3725927281/jobs/6319031966)

### PR DESCRIPTION
**Pull request by rsubtil**
Hopefully it fixes the current issues and generates less problems from now on.

Closes #293

### CODE REVIEW

**Comment by rsubtil** [comment]
Had to fix some issues in `cubinhos`, but they were long overdue anyways. I'm still not 100% on this, but it seems that defining the same `Options` struct on different files would confuse the program as to what's their intended destructor (ASAN reported errors because `Options` was constructed on `generate.cpp` and then deleted on `embed.cpp` for example). Giving them unique names per file seems to have fixed it.

File: engine/include/cubos/engine/cubos.hpp [hunk 1]
```diff
@@ -13,12 +13,14 @@ namespace cubos::engine
     /// Resource used as a flag to indicate whether the engine should stop running.
     struct ShouldQuit
     {
+        ShouldQuit(bool value) : value(value) {}
```
Comment (RiscadoA) [inline engine/include/cubos/engine/cubos.hpp:16]: This needs to be inlined or moved to the `.cpp` file.

File: engine/include/cubos/engine/cubos.hpp [hunk 2]
```diff
@@ -13,12 +13,14 @@ namespace cubos::engine
...
         bool value;
     };
 
     /// Resource used for storing program arguments.
     struct Arguments
     {
+        Arguments(const std::vector<std::string>& value) : value(value) {}
```
Comment (RiscadoA) [inline engine/include/cubos/engine/cubos.hpp:23]: This needs to be inlined or moved to the `.cpp` file.

**Review by RiscadoA — CHANGES_REQUESTED** [review]

File: engine/include/cubos/engine/cubos.hpp (hunk 1 above)
Comment (rsubtil) [reply engine/include/cubos/engine/cubos.hpp]: Fixed

File: engine/include/cubos/engine/cubos.hpp (hunk 2 above)
Comment (rsubtil) [reply engine/include/cubos/engine/cubos.hpp]: Fixed

**Review by RiscadoA — DISMISSED** [review]

**Review by RiscadoA — APPROVED** [review]

**Pull request merged**